pingcap/tidb
# *: add global variable tidb_schema_version_cache_limit to control infoschema cache size

### What problem does this PR solve?


Issue Number: close #46524

Problem Summary: add global variable tidb_schema_version_cache_limit to control infoschema cache size and add/refine metric.

### What is changed and how it works?

To fix #46524, we can simply enlarge the info-scheme cache capacity from 16 to 64. However, in scenarios with a large number of tables, this increases the risk of TiDB OOM.

So this PR just add global variable `tidb_schema_version_cache_limit ` to control infoschema cache size and refine metric in grafana to better diagnose this issue later. If you meet issue in #46524 with same root cause, you can enlarge the global variable `tidb_schema_version_cache_limit ` to fix it.

### Check List

Tests 


1. Deploy a TiDB cluster.
2. run stale read load.
```
sysbench --config-file=sysbench.conf mussel_minimal.lua --threads=20 --read_staleness=-5 run
```

3. run DDL load by [loadgen](https://github.com/crazycs520/loadgen). Following load will use 1 thread to continuously create tables, from t1 to t100000.

```
loadgen bench --sql 'create table t#seq-val (id int key, b int)' --thread=1 --valmin=1 --valmax=100000 --db=test2
```
4. check the metrics.

<img width="1445" alt="image" src="https://github.com/pingcap/tidb/assets/26020263/575582e0-06b7-4335-a760-a7d0cc6ece93">

- 17:12: Start running stale-read load, and the query QPS is 18000.
- 17:14: Start running DDL load, then the query QPS is dropping to 400. From the `Load Schema OPS` metrics, you can see many load snapshot schema OPS, this is what causes QPS to drop.
- 17:18: `set @@global. tidb_schema_version_cache_limit=64;`, then the query QPS return to 18000.



Side effects


Documentation

- N/A

### Release note



Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```
